src/repositories/powerBoostingRepository.ts

```diff
@@ -165,7 +165,11 @@ export const findPowerBoostingsCountByUserId = async (
     .leftJoin('powerBoosting.user', 'user')
     .addSelect(publicSelectionFields)
     .where(`percentage > 0`)
-    .andWhere(`"userId" =${userId}`);
+    .andWhere(`"userId" =${userId}`)
+    .cache(
```

Comment:
Are you sure it doesn't impact user experience? I wonder whether UI relies on this value for rendering modals or components, @MohammadPCh can you confirm or introduce a FE dev who is involved with this?
If it's called a lot and calculated by each user fetch, we may have to find another way around like removing it as a field and making a separate endpoint/resolver to fetch it.

Reply:
I agree we must either precalculate and save the data.
Or add cache.

We can't have queries executed everytime on the user on any minimal call.
Because basically 1 query to the user does 4 more under the hood @aminlatifi. So the complexity keeps growing.

Reply:
It was the exact same issue with the project queries that I added cache.

Reply:
@MohammadPCh @RamRamez can double check if those user values are required to be updated every second. I added a 1 minute cache.